builtin/credential/appgroup/path_app.go

```diff
@@ -0,0 +1,931 @@
+package appgroup
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/fatih/structs"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/vault/helper/policyutil"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+// appStorageEntry stores all the options that are set on an App
+type appStorageEntry struct {
+	// Policies that are to be required by the token to access the App
+	Policies []string `json:"policies" structs:"policies" mapstructure:"policies"`
+
+	// Number of times the UserID generated against the App can be used to perform login
+	NumUses int `json:"num_uses" structs:"num_uses" mapstructure:"num_uses"`
+
+	// Duration (less than the backend's mount) after which a UserID generated against the App will expire
+	UserIDTTL time.Duration `json:"userid_ttl" structs:"userid_ttl" mapstructure:"userid_ttl"`
+
+	// Duration before which an issued token should renew itself
+	TokenTTL time.Duration `json:"token_ttl" structs:"token_ttl" mapstructure:"token_ttl"`
+
+	// Duration after which an issued token should not be allowed to be renewed
+	TokenMaxTTL time.Duration `json:"token_max_ttl" structs:"token_max_ttl" mapstructure:"token_max_ttl"`
+
+	// If set, activates cubbyhole mode for the UserIDs generated against the App.
+	// An intermediary token will have the actual UserID reponse written in its cubbyhole.
+	// The value of Wrapped will be the duration after which the intermediary token
+	// along with its cubbyhole will be destroyed.
+	Wrapped time.Duration `json:"wrapped" structs:"wrapped" mapstructure:"wrapped"`
```

Comment:
In the `cubbyhole-the-world` branch this is called (internally) WrapTTL in the request, and the CLI flag is `-wrap-ttl`. Just using `wrapped` makes it seem like a bool, so we should probably sync this.